kubernetes/loculus/templates/silo-deployment.yaml

```
@@ -29,7 +29,8 @@ spec:
         {{- include "loculus.configProcessor" (dict "name" "lapis-silo-database-config" "dockerTag" $dockerTag) | nindent 8 }}
       containers:
         - name: silo
-          image: {{ $.Values.images.lapisSilo }}
+          image: "{{ $.Values.images.lapisSilo.repository }}:{{ $.Values.images.lapisSilo.tag | default "latest" }}"
```

```suggestion
          image: "{{ $.Values.images.lapisSilo.repository }}:{{ $.Values.images.lapisSilo.tag }}"
```
doesn't seem necessary? You could always configure `latest`

---

I think so at least

---

Well in that case you don't have one a bit below for `silo-preprocessing`. But I'm not really convinced. Someone could write a values.yaml with `tag: ""` and that would I guess yield something that doesn't start, but they could also write a values.yaml with `tag: "NonExistentTag"` which would also yield something that doesn't start

---

I guess it's a Q of whether we are seeing the schema as something to help external deployers or something to help us ensure our own starting `values.yaml` is compliant. I see it more as the former than the latter.

---

Hmm, I'm fine with leaving it out!

But I think I'm not following what you're saying. Yes, one could specify an invalid tag. But I think that's different than omitting the tag (as the schema suggests you're allowed to do) and then it not starting.

I'm also not following what the distinction would be you're making between the target of the schema. I think the way I think about it is that, if I use a key that the schema doesn't require, my template needs to account for that and take measures to produce valid output. since the tag can be left out, and not providing a default would create an invalid image string, I think we would need to provide a default here.

---

it's jsut that if you leave out the tag for the "website", the Chart will do something else and it will start. so it seemed to me that the same behaviour can be expected for the other images as well - hence the use of the default.

---

I'm saying that for an end-user-maintainer there isn't really an easy way they can omit the tag as it is specified in the base `values.yaml` from which everything is inherited.

---

To be clear my understanding is that if the user specifies a repository and not an tag, the LAPIS tag in the default values.yaml will be used

---

Ahhh I think I got it now. Yes the values.yaml is basically like a "default".

I'll go remove the marked bit.